<?xml version="1.0"?>
<launch>
    <arg name="configdirectory" default="/tmp"/>

    <include file="$(find roomba500)/launch/roombainterface.launch"/>

    <include file="$(find roomba500)/launch/imu.launch">
        <arg name="configdirectory" value="$(arg configdirectory)"/>
    </include>

    <node name="differentialodometry" pkg="roomba500" type="differentialodometryv2.py" output="screen">
        <param name="rosout_disable_topics_generation" value="True"/>
        <param name="ticksPerCm" value="22.7157014"/>
        <param name="robotWheelSeparationInCm" value="22.86"/>
        <param name="publish_tf" value="False" />
        <remap from="odom" to="wheelencoder/odom"/>
    </node>

    <node name="ekf_localization_fuser" pkg="robot_localization" type="ekf_localization_node" output="screen" respawn="false">
        <param name="frequency" value="20"/>
        <param name="two_d_mode" value="True"/>
        <param name="publish_tf" value="True" />

        <!-- The order of the boolean values are X,Y,Z,roll,pitch,yaw,X˙,Y˙,Z˙,roll˙,pitch˙,yaw˙,X¨,Y¨,Z¨. -->
        <param name="odom0" value="wheelencoder/odom"/>
        <rosparam param="odom0_config">[false, false, false,
            false, false, false,
            true, false, false,
            false, false, true,
            false, false, false]</rosparam>

        <param name="imu0" value="imu/data_raw"/>
        <rosparam param="imu0_config">[false, false, false,
            false, false, true,
            false, false, false,
            false, false, true,
            false, false, false]</rosparam>
        <param name="imu0_differential" value="True" />

        <param name="imu1" value="imu/compass_orientation"/>
        <rosparam param="imu1_config">[false, false, false,
            false, false, true,
            false, false, false,
            false, false, false,
            false, false, false]</rosparam>

        <remap from="odometry/filtered" to="odom"/>
    </node>

    <include file="$(find roomba500)/launch/lidar.launch">
        <arg name="configdirectory" value="$(arg configdirectory)"/>
    </include>
</launch>